### PR TITLE
Dwarfvet: third time is the charm

### PR DESCRIPTION
**Pull request by NCommander**
Actually got the PR right this time.

Implements dwarfvet which allows animal caretakers to work.  Original PR is here https://github.com/DFHack/dfhack/pull/937  before I moved it to its own branch.

### CODE REVIEW

File: NEWS.rst [hunk 1]
```diff
@@ -40,6 +40,7 @@ Lua
 New Features
 ------------
 - `gui/gm-editor` it's now possible to insert default types to containers. For primitive types leave the type entry empty, for references use ``*``.
+- `dwarfvet` enables animal caretaking.
```
Comment (lethosor) [inline NEWS.rst]: Should be under "new plugins" (which probably doesn't exist, but see the initial comment or previous releases for where it goes).

File: docs/Plugins.rst [hunk 2]
```diff
@@ -1109,6 +1109,27 @@ Example:
     This will confiscate rotten and dropped food, garbage on the floors and any
     worn items with 'X' damage and above.
 
+.. _dwarfvet:
```
Comment (lethosor) [inline docs/Plugins.rst]: I think this file is sorted alphabetically, so this should be after dwarfmonitor.

File: plugins/CMakeLists.txt [hunk 3]
```diff
@@ -93,6 +93,7 @@ if (BUILD_SUPPORTED)
     DFHACK_PLUGIN(dig dig.cpp)
     DFHACK_PLUGIN(digFlood digFlood.cpp)
     add_subdirectory(diggingInvaders)
+	DFHACK_PLUGIN(dwarfvet dwarfvet.cpp)
```
Comment (lethosor) [inline plugins/CMakeLists.txt]: Indentation doesn't match - probably change this to spaces.

File: docs/Plugins.rst (hunk 2 above)
Comment (NCommander) [inline docs/Plugins.rst]: It's not at least according to the index on docs, but I reordered this.

**Pull request merged**